includes/functions/plugins/article-export.php

```diff
@@ -459,30 +545,170 @@ function build_xml() {
 			)
 		);
 
-		//perform dom manipulations
+		// perform dom manipulations
 		$dom = $this->manipulate_dom( $dom );
 
-		//do html_to_xml before adding elements so that the html wrap stuff is removed first
+		// do html_to_xml before adding elements so that the html wrap stuff is removed first
 		$xml_elements = $this->html_to_xml( $dom );
 
+		$elements = apply_filters(
+			__NAMESPACE__ . '\xml_outer_elements',
+			$this->get_outer_elements( $this->article ),
+			$this->article
+		);
+
 		if ( $elements ) {
 			$this->add_outer_elements( $xml_elements, $elements );
 		}
 
 		$this->add_article_attributes( $xml_elements->root_element, $elements );
 
-		//reset global post
+		// reset global post
 		$post = $old;
 
+		$GLOBALS['recentComment'] = $elements['comment'] ? $elements['comment'] : false;
+
 		return $xml_elements;
+	}
+
+	/**
+	 * Get prepared article content
+	 *
+	 * @param object $article WP_Post article
+	 *
+	 * @return string article content
+	 */
+	function get_article_content( $article ) {
+
+		$content = $this->article->post_content;
+
+		if ( $post = get_post( get_post_thumbnail_id( $this->id ) ) ) {
+			if ( $image = $this->get_image_name( $post->ID ) ) {
+				$content = $this->get_image_tag( $image[1], $post->post_excerpt ) . $content;
+			}
+		}
+
+		$content = preg_replace_callback(
+			'/\[caption.*\[\/caption\]/Usi',
+			function ( $matches ) {
+				if ( preg_match( '/attachment_(\d+)\D/i', $matches[0], $matches2 ) ) {
+					$image = $this->get_image_name( (int) $matches2[1] );
+				}
+				if ( preg_match( '/\[caption[^\]]*](.*)\[\/caption\]/Usi', strip_tags( $matches[0] ), $matches2 ) ) {
+					$caption = trim( $matches2[1] );
+				}
+				return $image ? $this->get_image_tag( $image[1], $caption ) : '';
+			},
+			$content
+		);
+
+		$content = preg_replace_callback(
+			'/\[gallery[^]]*ids="([\d,]+)"[^]]*\]/Usi',
+			function ( $matches ) {
+
+				if ( $matches[1] ) {
+					$gallery_images = explode( ',', $matches[1] );
+					foreach ( $gallery_images as $img_id ) {
+						$image = $this->get_image_name( (int) $img_id );
+						wp_mail( 'aaaa@asefetbhtregfsefwef.com', 'TEST', print_r( $image, 1 ) );
```

Comment:
The test code should be removed.

Reply:
Done, same as https://github.com/10up/eight-day-week/pull/117#discussion_r1241271756. Was there previously.